portfolio-project-matching-app/models/Project.js

```diff
@@ -1,20 +1,88 @@
 class Project {
+    // fields
+    id;
+    name;
+    description;
+    capacity;
+    census;
+    open;
+    likes;
+    ownerId;
+    owner;
+    technologies;
+    users;
 
-    constructor(docSnapshot) {
-        this.id = docSnapshot.id;
-        this.name = docSnapshot.data().name;
-        this.description = docSnapshot.data().description;
-        this.capacity = docSnapshot.data().capacity;
-        this.census = docSnapshot.data().census;
-        this.open = docSnapshot.data().open;
-        this.likes = docSnapshot.data().likes;
-        this.ownerId = docSnapshot.data().owner.id;
-        this.owner = null;
-        this.technologies = null;
-        this.users = null;
+    // static factory methods (used like overloaded constructors)
+    static fromDocSnapshot(id, docSnapshot) {
```

Comment:
I added some static factory methods in place of having a single constructor. This gives the developer a bit more freedom when creating one of our custom objects / classes.